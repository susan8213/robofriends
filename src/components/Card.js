import React from 'react';
import "tachyons";

const Card = ({ id, name, email }) => {
    return (
        <div className="br3 pa3 ma2 tc bg-lightest-blue bw2 ba b--black-10 dib grow">
            <img 
                src={`https://robohash.org/${id}?set=set4&size=250x250`}
                alt="Cat" 
                className="br-100 h4 w4 dib pa2"
            ></img>
            <h1 className="f3 mb2">{name}</h1>
            <h2 className="f5 fw4 dark-gray mt0">{email}</h2>
        </div>
    )
}

export default Card;