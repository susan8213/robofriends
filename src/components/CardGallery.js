import React from "react";
import Card from "./Card";

const CardGallery = ({ robots }) => {
    return (
        <React.Fragment>
            {robots.map(user => (
                <Card key={user.id} 
                    id={user.id} 
                    name={user.name} 
                    email={user.email} 
                />))}
        </React.Fragment>
    )
    
}

export default CardGallery;