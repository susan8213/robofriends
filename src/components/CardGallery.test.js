import { shallow } from 'enzyme';
import React from 'react';
import CardGallery from './CardGallery';
import Card from './Card';


describe('<CardGallery />', () =>{

    it('render without crashing', () => {
        expect(shallow(<CardGallery robots={[]}/>).html()).toMatchSnapshot();
    })

    it('should contain <Card />', () => {
        const mockRobots = [{id: 1}];
        const expectedChild = <Card id={1}/>;
        expect(shallow(<CardGallery robots={mockRobots}/>).contains(expectedChild)).toBe(true);
    })


})