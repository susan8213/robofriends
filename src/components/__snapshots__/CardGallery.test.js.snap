// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardGallery /> render without crashing 1`] = `""`;
